qemu/tests/virtio_fs_group_permission_access.py

```diff
@@ -51,8 +51,10 @@ def run(test, params, env):
 
     try:
         for _username in username:
-            cmd = add_user_cmd % _username
-            process.run(cmd)
+            if process.system("id %s" % _username, shell=True,
+                              ignore_status=True) == 0:
+                process.run(del_user_cmd % _username)
```

Comment:
Here may be failed to delete user?? If NOT, the code is okay for me.
Otherwise, use the follow way to delete the user: https://github.com/autotest/tp-qemu/blob/78e23feddac79f9844aafb2f7ebd181013a5e936/qemu/tests/virtio_fs_group_permission_access.py#L143

However, based on my understanding on this test step, "delete the user is failed" does NOT happen at most time.
 Do we need this way to delete user? @YongxueHong @zhencliu

Reply:
From my understanding, at the first of the testing, we try to clean up the env(delete users) first in case the users are not deleted in the previous testings, so the key point here is why we still have the users? why we don't(or fail to) delete them in previous cases? It's better to do the cleanup work after each case, then we don't have to make a double cleanup in the beginning of each case.

Another point, if 'delete users' is not part of our testing(ie. it's not a testing assertion), we can backup the image in the beginning, and delete the image after testing, finally, restore the image from its backup, simple and easy.

Reply:
> Here may be failed to delete user?? If NOT, the code is okay for me. Otherwise, use the follow way to delete the user: https://github.com/autotest/tp-qemu/blob/78e23feddac79f9844aafb2f7ebd181013a5e936/qemu/tests/virtio_fs_group_permission_access.py#L143-#L148
> 
> However, based on my understanding on this test step, "delete the user is failed" does NOT happen at most time. Do we need this way to delete user? @YongxueHong @zhencliu
Hi @nickzhq 

Yeah, I also think that it does not happen most time if the env is cleaned up completely after each testing case, but there are still limited possibilities here that some cases do not clean the environment well so it will lead to the next failure.

So, I think we could double-check it even if it may dry run most time, but we could keep the following cases run with a pure test environment.

Please let me know your opinions. Thanks.

Reply:
> From my understanding, at the first of the testing, we try to clean up the env(delete users) first in case the users are not deleted in the previous testings, so the key point here is why we still have the users? why we don't(or fail to) delete them in previous cases? It's better to do the cleanup work after each case, then we don't have to make a double cleanup in the beginning of each case.
> 
> Another point, if 'delete users' is not part of our testing(ie. it's not a testing assertion), we can backup the image in the beginning, and delete the image after testing, finally, restore the image from its backup, simple and easy.

Hello @zhencliu , The "delete user" should be executed on host instead of guest, "backup the image" is NOT helpful for this issue.

Reply:
> > Here may be failed to delete user?? If NOT, the code is okay for me. Otherwise, use the follow way to delete the user: https://github.com/autotest/tp-qemu/blob/78e23feddac79f9844aafb2f7ebd181013a5e936/qemu/tests/virtio_fs_group_permission_access.py#L143-#L148
> > However, based on my understanding on this test step, "delete the user is failed" does NOT happen at most time. Do we need this way to delete user? @YongxueHong @zhencliu
> > Hi @nickzhq
> 
> Yeah, I also think that it does not happen most time if the env is cleaned up completely after each testing case, but there are still limited possibilities here that some cases do not clean the environment well so it will lead to the next failure.
> 
> So, I think we could double-check it even if it may dry run most time, but we could keep the following cases run with a pure test environment.
> 
> Please let me know your opinions. Thanks.

I prefer to keep the double clean, but the clean way should be like this: https://github.com/autotest/tp-qemu/blob/78e23feddac79f9844aafb2f7ebd181013a5e936/qemu/tests/virtio_fs_group_permission_access.py#L143-#L148

cc @xiagao @zhencliu

Reply:
> > > Here may be failed to delete user?? If NOT, the code is okay for me. Otherwise, use the follow way to delete the user: https://github.com/autotest/tp-qemu/blob/78e23feddac79f9844aafb2f7ebd181013a5e936/qemu/tests/virtio_fs_group_permission_access.py#L143-#L148
> > > However, based on my understanding on this test step, "delete the user is failed" does NOT happen at most time. Do we need this way to delete user? @YongxueHong @zhencliu
> > > Hi @nickzhq
> > 
> > 
> > Yeah, I also think that it does not happen most time if the env is cleaned up completely after each testing case, but there are still limited possibilities here that some cases do not clean the environment well so it will lead to the next failure.
> > So, I think we could double-check it even if it may dry run most time, but we could keep the following cases run with a pure test environment.
> > Please let me know your opinions. Thanks.
> 
> I prefer to keep the double clean, but the clean way should be like this: https://github.com/autotest/tp-qemu/blob/78e23feddac79f9844aafb2f7ebd181013a5e936/qemu/tests/virtio_fs_group_permission_access.py#L143-#L148
> 
> cc @xiagao @zhencliu

It looks good to me, normally, the step for cleaning should be done at last if each case follows up on this suggestion.

Hi @xiagao 
Would you like to give more details about the issue?
I am a bit curious about what kind of reason results in such an issue. 
Thanks.

Reply:
> > From my understanding, at the first of the testing, we try to clean up the env(delete users) first in case the users are not deleted in the previous testings, so the key point here is why we still have the users? why we don't(or fail to) delete them in previous cases? It's better to do the cleanup work after each case, then we don't have to make a double cleanup in the beginning of each case.
> > Another point, if 'delete users' is not part of our testing(ie. it's not a testing assertion), we can backup the image in the beginning, and delete the image after testing, finally, restore the image from its backup, simple and easy.
> 
> Hello @zhencliu , The "delete user" should be executed on host instead of guest, "backup the image" is NOT helpful for this issue.

Thanks for the reminder, I didn't notice that, then I am curious why we fail to delete a user from the host, it may be valuable to dive into this, my consideration: we finish our testing, stop qemu/virtio-fs daemon(maybe?), and delete the users, it should be done without any issue.  Maybe we need to check the loop and find out what the previous case is, then find if we didn't follow a right way to do the cleanup work , if we did the cleanup work  in a proper way, then I doubt a bug here, i.e. some resource is busy using by the users,  just FYI, cc @xiagao

For this case, it's OK for me to do a initial cleanup. cc @nickzhq